addons/webhooks/suite_test.go

```diff
@@ -0,0 +1,167 @@
+package webhooks_test
+
+import (
+	"path/filepath"
+	"testing"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	adminregv1 "k8s.io/api/admissionregistration/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	clusterapiv1beta1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	controlplanev1beta1 "sigs.k8s.io/cluster-api/controlplane/kubeadm/api/v1beta1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/envtest"
+	"sigs.k8s.io/controller-runtime/pkg/envtest/printer"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+
+	kappctrl "github.com/vmware-tanzu/carvel-kapp-controller/pkg/apis/kappctrl/v1alpha1"
+	pkgiv1alpha1 "github.com/vmware-tanzu/carvel-kapp-controller/pkg/apis/packaging/v1alpha1"
+	"github.com/vmware-tanzu/tanzu-framework/addons/testutil"
+	cniv1alpha1 "github.com/vmware-tanzu/tanzu-framework/apis/cni/v1alpha1"
+	runtanzuv1alpha1 "github.com/vmware-tanzu/tanzu-framework/apis/run/v1alpha1"
+	runtanzuv1alpha3 "github.com/vmware-tanzu/tanzu-framework/apis/run/v1alpha3"
+)
+
+const (
+	ServiceName  = "webhook-service"
+	NameSpace    = "tkg-system"
+	CertDirPath  = "/tmp/webhookserver"
+	CertFileName = "tls.crt"
+	KeyFileName  = "tls.key"
+)
+
+var (
+	cfg       *rest.Config
+	testEnv   *envtest.Environment
+	scheme    = runtime.NewScheme()
+	k8sClient client.Client
+	ctx       = ctrl.SetupSignalHandler()
+	fakeMgr   *manager.Manager
+)
+
+func TestWebhooks(t *testing.T) {
+	RegisterFailHandler(Fail)
+
+	RunSpecsWithDefaultAndCustomReporters(t,
+		"Addon Webhooks Suite",
+		[]Reporter{printer.NewlineReporter{}})
+}
+
+var _ = BeforeSuite(func(done Done) {
+
+	By("bootstrapping test environment")
```

Comment:
question: I think `--admission-control=` needs to be added to `envtest.DefaultKubeAPIServerFlags` for test setup.

Reply:
fixed this thanks!